libcxx/test/std/ranges/range.adaptors/range.chunk/adaptor.pass.cpp

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++23
+
+// std::views::chunk
+
+#include <ranges>
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <span>
+
+#include "test_range.h"
+#include "types.h"
+
+constexpr bool test() {
+  std::array array    = {1, 1, 1, 2, 2, 2, 3, 3};
+  std::span<int> span = {array.data(), 8};
+
+  // Test `views::chunk(view, n)`
+  {
+    auto chunked = std::views::chunk(span, 2);
+    static_assert(std::same_as<decltype(chunked), std::ranges::chunk_view<std::span<int>>>);
```

Comment:
The preferred pattern to test the return type is: `std::same_as<Result> decltype(auto) x = f();`, see: https://github.com/llvm/llvm-project/blob/0744da9e5af8545068fa157d4bce63c2d2436e2a/libcxx/test/std/ranges/range.adaptors/range.take.while/adaptor.pass.cpp#L58

e.g.:

```suggestion
    std::same_as<std::ranges::chunk_view<std::span<int>>>> decltype(auto) chunked = std::views::chunk(span, 2);
```

Also AFAIK it is preferred to specify the type explicitly instead of using `auto` if possible.

Reply:
Ok.